---
title: "Demo"
author: "Richel J.C. Bilderbeek"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Demo}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Usage

From a protein sequence, `bbbq` estimates where amino acids of transmembrane proteins 
are located (inside, outside, in the membrane) and which bind to an MHC2
allele with a certain strength.

We need a FASTA file with at least one protein sequence in it to work on:

```{r}
fasta_filename <- system.file("extdata", "short.fasta", package = "bbbq")
```

This is how such a FASTA file looks like:

```{r}
cat(head(readLines(fasta_filename, warn = FALSE)), sep = "\n")
```

Different MHC2 alleles bind differently to protein epitopes. 
By default, `bbbq` uses only the default MHC2 allele used by NetMCHIIpan.
In this demo, we'll use the first two MHC2 alleles from the complete
NetMHCIIpan set of more than 5000 alleles:

```{r}
alleles <- netmhc2pan::get_netmhc2pan_alleles()[1:2]
testit::assert(all(alleles %in% netmhc2pan::get_netmhc2pan_alleles()))
```

Select a binding strength. For example, a value of `5.0` will select
those epitopes that are in the top 5 percent.

```{r}
binding_strength_threshold <- 5.0
```

Here, the BBBQ is answered:

```{r}
df <- bbbq::answer_bbbq_1(
  fasta_filename = fasta_filename,
  alleles = alleles,
  binding_strength_threshold = binding_strength_threshold
)
```

Resulting in:

```{r}
knitr::kable(df)
```

Legend:

Location|Strong binder|Weak binder
---|---|---
outside|O|o
membrane|M|m
inside|I|i

## Appendix

This appendix shows the intermediate files created by `bbbq`.

These are:

 * `tmhmm_filename`: the TMHMM results file, containing the location of the
   amino acids (inside, outside, in membrane)
 * `netmhc2pan_filename`: the NetMHC2pan results file, containing the
   binding of all MHC2 alleles to all protein epitopes
 * `epitopeome_filename`: the `epitopeome` results file, containing the
   combinated informationb

```{r}
tmhmm_filename = tempfile(".txt")
netmhc2pan_filename = tempfile(".csv")
epitopeome_filename = tempfile(".fasta")
```

Answering the BBBQ again (and ignorening its result):

```{r}
bbbq::answer_bbbq_1(
  fasta_filename = fasta_filename,
  alleles = alleles,
  binding_strength_threshold = binding_strength_threshold,
  tmhmm_filename = tmhmm_filename,
  netmhc2pan_filename = netmhc2pan_filename,
  epitopeome_filename = epitopeome_filename
)
```

The TMHMM results file:

```{r}
cat(readLines(tmhmm_filename), sep = "\n")
```

The NetMHCIIpan results table:

```{r}
knitr::kable(read.csv(netmhc2pan_filename))
```

The `epitopeome` results file:

```{r}
cat(readLines(epitopeome_filename), sep = "\n")
```
