#' This function does nothing. It is intended to inherit is parameters'
#' documentation.
#' @param binding_strength_threshold threshold of the MHC2 allele's
#'   binding strength. A low threshold, e.g. \code{5.0}, selects for the top
#'   5 percent best binders
#' @param data_folder folder where BBBQ stores its data, such as proteomes
#' and prediced topologies.
#' @param epitopeome_filename name of the file to store the epitopeome
#'   output.
#'   By default, this filename is generated by \link{tempfile},
#'   and will be put in the \code{/tmp} folder,
#'   which is cleaned by the operating system
#' @param fasta_filename name of a FASTA file
#' @param haplotype one MHC-I or MHC-II haplotype
#' @param haplotypes a list of MHC-I or MHC-II haplotypes
#' @param hydrophobe_control_peptides_as_data_filename
#'   filename that stores hydrophobe control peptides
#'   as R data,
#'   for example `work/hydrophobe-control-peptides.Rdata`
#' @param ic50_prediction_tool tool to predict the IC50 from a
#'   peptide. Possible values are:\cr
#' \itemize{
#'   \item mhcnuggetsr \link[mhcnuggetsr]{mhcnuggetsr},
#'     which uses MHCnuggets
#'   \item mhcnuggetsr \link[netmhc2pan]{netmhc2pan},
#'     which uses NetMHC2pam
#'   \item EpitopePrediction uses \code{EpitopePrediction}
#' }
#' @param ic50_threshold peptides with a predicted IC50 value (in nM)
#'   lower than this will be considered binders.
#'   A threshold of 500 (nM) is common.
#' @param keep_selenoproteins keep the selenoproteins.
#' The reason to discard these, is that TMHMM cannot deal
#' with selenoproteins.
#' @param kyte_doolittle_scale_as_data_filename
#'   filename that stores the Kyte-Doolittle scale of
#'   hydrophobicity,
#'   for example `data/kyte.doolittle.scale.Rdata`
#' @param max_hydrophobicity the maximal hydrophobicity
#' @param mhc_class the MHC class, for example \code{1}
#' @param mhc_haplotype haplotype of either HMC-I or MHC-II.
#'   Use \link{get_mhc1_haplotypes} to get a list of all MHC-I haplotypes.
#'   Use \link{get_mhc2_haplotypes} to get a list of all MHC-II haplotypes.
#' @param mhc_1_haplotype HMC-I haplotope.
#'   Use \link{get_mhc1_haplotypes} to get a list of all haplotypes
#' @param mhc_2_haplotype HMC-II haplotope.
#'   Use \link{get_mhc2_haplotypes} to get a list of all haplotypes
#' @param mhcnuggetsr_peptides_path path to a temporary file to create
#'   peptides, as needed by \link{mhcnuggetsr},
#'   which is \link[mhcnuggetsr]{create_temp_peptides_path} by default
#' @param min_hydrophobicity the minimal hydrophobicity
#' @param msa_method the \code{method} argument of \link[msa]{msa}.
#' Valid methods are \code{ClustalW}, \code{ClustalOmega} (preferred for proteins)
#' and \code{Muscle}
#' @param msa_subst_matrix multiple sequence alignment substitution matrix,
#' the \code{substitutionMatrix} argument of \link[msa]{msa}.
#' For a protein alignment, relevant values are \code{blosum},
#' \code{pam} and \code{gonnet}
#' @param n_peptides number of peptides
#' @param non_tmh_eluted_filename
#'   Filename for the non-TMHs found on B-cells,
#'   for example `data/non-TMH-Bcell-elution.txt`
#' @param peptide peptide sequence, for example,
#' \code{ARNDCQEV}
#' @param peptides one ore more peptide sequences
#' @param peptide_length length of the peptide in amino acids
#' @param percentile how low the IC50 must be for the protein to
#'   be considered a binder. For example, 0.02 denotes that the protein
#'   must have an IC50 in the lowest 2 percent range.
#'   The default value is returned by \link{get_ic50_percentile_binder}.
#' @param png_filename name of a PNG file
#' @param protein_lengths_filename filename to store the
#'   length of all proteins in proteome,
#'   for example `work/protein-lengths.txt`
#' @param protein_sequence a protein sequence
#' @param protein_sequences one or more protein sequences
#' @param proteome_as_data_filename filename to store the
#'   proteome in R data format,
#'   for example `work/proteome.Rdata`
#' @param proteome_9mer_hydrophobicity_as_data_filename
#'   filename to store the
#'   proteome 9-mer hydrophobicities in R data format,
#'   for example `work/proteome.9mer.hydrophobicity.Rdata`
#' @param proteome_filename name of the file the proteome will be stored at
#' @param proteome_type type of proteome, must be
#' either \code{full} or \code{representative}
#' @param target_name abbreviated name of organism or virus.
#' Use \link{get_target_names} to get the valid values.
#' Use \link{get_target_english_name} to get the English name of a target.
#' @param tmh_9mers_as_data_filename filename to store
#'   per TMH protein, the indices at which it is TMH,
#'   in R data format.
#'   for example `work/tmh.9mers.Rdata`
#' @param tmh_eluted_filename
#'   Filename for the TMHs found on B-cells,
#'   for example, `data/TMH-Bcell-elution.txt`
#' @param tmh_overlapping_binders_as_data_filename filename to store
#'   per TMH overlapping binders
#'   in R data format.
#'   for example `work/tmh-overlapping-binders.Rdata`
#' @param transition_matrix the amino acid transition matrix,
#'   for example, the BLOSUM62 transition matrix
#'   as returned by \link{get_aa_transition_matrix}
#' @param transition_matrix_name the name of an amino acid transition matrix,
#'   for example, \code{BLOSUM62}
#' @param pureseqtmr_filename name of the file to store the
#'   PureseqTM output.
#'   By default, this filename is generated by \link{tempfile},
#'   and will be put in the \code{/tmp} folder,
#'   which is cleaned by the operating system
#' \code{mhcn} for \link[mhcnuggetsr]{mhcnuggetsr}
#' @param topology a topology,
#'   as created by \link[pureseqtmr]{predict_topology}
#' @param topology_prediction_tool tool to predict the topology
#'   of a membrane protein. Possible values are:\cr
#' \itemize{
#'   \item mhcnuggetsr \link[pureseqtmr]{pureseqtmr},
#'     which uses PureseqTM
#'   \item mhcnuggetsr \link[tmhmm]{tmhmm},
#'     which uses TMHMM
#' }
#' @param trans_membrane_analysis_filename Filename for
#'   uhhh, something, for example
#'   `tmh-predictions/trans-membrane-analysis-shortened.txt`
#' @param triplet a DNA triplet
#' @param verbose set to TRUE for more output
#' @author Rich√®l J.C. Bilderbeek
#' @note This is an internal function, so it should be marked with
#'   \code{@noRd}. This is not done, as this will disallow all
#'   functions to find the documentation parameters
default_params_doc <- function(
  binding_strength_threshold,
  data_folder,
  epitopeome_filename,
  fasta_filename,
  haplotype,
  haplotypes,
  hydrophobe_control_peptides_as_data_filename,
  ic50_prediction_tool,
  ic50_threshold,
  keep_selenoproteins,
  kyte_doolittle_scale_as_data_filename,
  max_hydrophobicity,
  mhc_class,
  mhc_haplotype,
  mhc_1_haplotype,
  mhc_2_haplotype,
  mhcnuggetsr_peptides_path,
  min_hydrophobicity,
  msa_method,
  msa_subst_matrix,
  n_peptides,
  non_tmh_eluted_filename,
  peptide,
  peptides,
  peptide_length,
  percentile,
  png_filename,
  protein_lengths_filename,
  protein_sequence,
  protein_sequences,
  proteome_filename,
  proteome_as_data_filename,
  proteome_9mer_hydrophobicity_as_data_filename,
  proteome_type,
  pureseqtmr_filename,
  target_name,
  tmh_9mers_as_data_filename,
  tmh_eluted_filename,
  tmh_overlapping_binders_as_data_filename,
  topology,
  topology_prediction_tool,
  transition_matrix,
  transition_matrix_name,
  trans_membrane_analysis_filename,
  triplet,
  verbose
) {
  # Nothing
}
