#' This function does nothing. It is intended to inherit is parameters'
#' documentation.
#' @param alleles one or more alleles, e.g. \code{DRB1_0101}.
#'   See \link[netmhc2pan]{get_netmhc2pan_alleles} for a full list
#' @param binding_strength_threshold threshold of the MHC2 allele's
#'   binding strength. A low threshold, e.g. \code{5.0}, selects for the top
#'   5 percent best binders
#' @param epitopeome_filename name of the file to store the epitopeome
#'   output.
#'   By default, this filename is generated by \link{tempfile},
#'   and will be put in the \code{/tmp} folder,
#'   which is cleaned by the operating system
#' @param fasta_filename name of a FASTA file
#' @param haplotypes a list of MHC-I or MHC-II haplotypes
#' @param hydrophobe_control_peptides_as_data_filename
#'   filename that stores hydrophobe control peptides
#'   as R data,
#'   for example `work/hydrophobe-control-peptides.Rdata`
#' @param kyte_doolittle_scale_as_data_filename
#'   filename that stores the Kyte-Doolittle scale of
#'   hydrophobicity,
#'   for example `data/kyte.doolittle.scale.Rdata`
#' @param n_aas number of amino acids
#' @param netmhc2pan_filename name of the file to store the
#'   NetMHC2pan output.
#'   By default, this filename is generated by \link{tempfile},
#'   and will be put in the \code{/tmp} folder,
#'   which is cleaned by the operating system
#' @param non_tmh_eluted_filename
#'   Filename for the non-TMHs found on B-cells,
#'   for example `data/non-TMH-Bcell-elution.txt`
#' @param png_filename name of a PNG file
#' @param protein_lengths_filename filename to store the
#'   length of all proteins in proteome,
#'   for example `work/protein-lengths.txt`
#' @param protein_sequence a protein sequence
#' @param protein_sequences one or more protein sequences
#' @param proteome_as_data_filename filename to store the
#'   proteome in R data format,
#'   for example `work/proteome.Rdata`
#' @param proteome_9mer_hydrophobicity_as_data_filename
#'   filename to store the
#'   proteome 9-mer hydrophobicities in R data format,
#'   for example `work/proteome.9mer.hydrophobicity.Rdata`
#' @param proteome_filename name of the file the proteome will be stored at
#' @param tmh_9mers_as_data_filename filename to store
#'   per TMH protein, the indices at which it is TMH,
#'   in R data format.
#'   for example `work/tmh.9mers.Rdata`
#' @param tmh_eluted_filename
#'   Filename for the TMHs found on B-cells,
#'   for example, `data/TMH-Bcell-elution.txt`
#' @param tmh_overlapping_binders_as_data_filename filename to store
#'   per TMH overlapping binders
#'   in R data format.
#'   for example `work/tmh-overlapping-binders.Rdata`
#' @param tmhmm_filename name of the file to store the
#'   TMHMM output.
#'   By default, this filename is generated by \link{tempfile},
#'   and will be put in the \code{/tmp} folder,
#'   which is cleaned by the operating system
#' @param trans_membrane_analysis_filename Filename for
#'   uhhh, something, for example
#'   `tmh-predictions/trans-membrane-analysis-shortened.txt`
#' @param triplet a DNA triplet
#' @param verbose set to TRUE for more output
#' @author Richel J.C. Bilderbeek
#' @note This is an internal function, so it should be marked with
#'   \code{@noRd}. This is not done, as this will disallow all
#'   functions to find the documentation parameters
default_params_doc <- function(
  alleles,
  binding_strength_threshold,
  epitopeome_filename,
  fasta_filename,
  haplotypes,
  hydrophobe_control_peptides_as_data_filename,
  kyte_doolittle_scale_as_data_filename,
  n_aas,
  netmhc2pan_filename,
  non_tmh_eluted_filename,
  png_filename,
  protein_lengths_filename,
  protein_sequence,
  protein_sequences,
  proteome_filename,
  proteome_as_data_filename,
  proteome_9mer_hydrophobicity_as_data_filename,
  tmh_9mers_as_data_filename,
  tmh_eluted_filename,
  tmh_overlapping_binders_as_data_filename,
  tmhmm_filename,
  trans_membrane_analysis_filename,
  triplet,
  verbose
) {
  # Nothing
}
